client/client_test.go

```diff
@@ -0,0 +1,190 @@
+package client
+
+import (
+	"context"
+	"errors"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/rs/zerolog"
+)
+
+func TestNew(t *testing.T) {
+	logger := zerolog.New(os.Stdout)
+	ctx := context.Background()
+
+	tests := []struct {
+		name        string
+		spec        *Spec
+		wantErr     bool
+		expectedErr error
+	}{
+		{
+			name:        "missing org",
+			spec:        &Spec{},
```

Comment:
Could we reduce the space for unexpected behaviour by providing all fields except the org? That way, this test is less likely to break in the future if the validation logic changes.